alacritty/src/message_bar.rs

```diff
@@ -55,16 +56,24 @@ impl Message {
                 if let (Some(index), true) = (line.rfind(char::is_whitespace), c != '\n') {
                     let split = line.split_off(index + 1);
                     line.pop();
+
                     lines.push(Self::pad_text(line, num_cols));
-                    line = split
+                    line = split;
+                    line_len = line.width();
                 } else {
                     lines.push(Self::pad_text(line, num_cols));
                     line = String::new();
+
+                    line_len = 0;
                 }
             }
 
             if c != '\n' {
                 line.push(c);
+
+                if let Some(char_width) = c.width() {
+                    line_len += char_width;
+                }
```

Comment:
Should just use `.unwrap_or(0)`. Might have to use `.unwrap_or_default()` if clippy complains, but `unwrap_or(0)` is more descriptive and shorter so I think it's better in this case.